pkg/nodes/node.go

```diff
@@ -189,6 +196,94 @@ func (builder *Builder) ExternalIPv4Network() (string, error) {
 	return extNetwork.IPv4, nil
 }
 
+// IsReady check if the Node is Ready.
+func (builder *Builder) IsReady() (bool, error) {
+	if valid, err := builder.validate(); !valid {
+		return false, err
+	}
+
+	glog.V(100).Infof("Verify %s node availability", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return false, fmt.Errorf("%s node object doesn't exist", builder.Definition.Name)
+	}
+
+	for _, condition := range builder.Object.Status.Conditions {
+		if condition.Type == "Ready" {
+			return condition.Status == isTrue, nil
+		}
+	}
+
+	return false, fmt.Errorf("the Ready condition could not be found for node %s", builder.Definition.Name)
+}
+
+// WaitUntilConditionTrue waits for timeout duration or until node gets to a specific status.
+func (builder *Builder) WaitUntilConditionTrue(
+	conditionType v1.NodeConditionType, timeout time.Duration) error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	err := wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		if !builder.Exists() {
+			return false, fmt.Errorf("node %s object doesn't exist", builder.Definition.Name)
+		}
+
+		for _, condition := range builder.Object.Status.Conditions {
+			if condition.Type == conditionType {
+				return condition.Status == "True", nil
+			}
+		}
+
+		return false, fmt.Errorf("%s node is not in %s state", builder.Definition.Name, conditionType)
+	})
+
+	if err == nil {
+		return nil
+	}
+
+	return fmt.Errorf("%s node is not in %s state due to %w", builder.Definition.Name, conditionType, err)
+}
+
+// WaitUntilConditionFalse waits for timeout duration or until node change specific status.
+func (builder *Builder) WaitUntilConditionFalse(
+	conditionType v1.NodeConditionType, timeout time.Duration) error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	err := wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		if !builder.Exists() {
+			return false, fmt.Errorf("node %s object doesn't exist", builder.Definition.Name)
+		}
+
+		for _, condition := range builder.Object.Status.Conditions {
+			if condition.Type == conditionType {
+				return condition.Status == "Unknown", nil
```

Comment:
Why "Unknown" instead of "False"?

Reply:
cause there is no False, but Unknown :)

Reply:
I think we should change this to WaitUntilConditionUnknown and use that for you reboot check. The False status does exist on conditions while the node is booted and reporting correctly.

Reply:
I'm not so sure about it, it can confuse. Externally (oc get nodes) status called as NotReady